readthedocs/search/parsers.py

```diff
@@ -123,6 +123,59 @@ def _parse_sections(self, title, body):
         We can have pages that have content before the first title or that don't have a title,
         we index that content first under the title of the original page.
         """
+
+        # All terms in dls are treated as sections.
+        # After indexing them, we remove them from the body such that their contents aren't
+        # repeated in the following indexing
+        dls = body.css("dl")
+        seen_dls = []
+        for dl in dls:
+            if dl in seen_dls:
+                continue
+            for inner_dl in dl.css("dl"):
+                seen_dls.append(inner_dl)
+
+            # Select all dts with id defined
+            dts = dl.css('dt[id]:not([id=""])')
+
+            # https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt
+            # multiple <dt> elements in a row indicate several terms that are
+            # all defined by the immediate next <dd> element.
+            for dt in dts:
+                try:
+                    title, _id = self._parse_dt(dt)
+                    # https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator
+                    dd = dt.css_first("dt ~ dd")
+                    # We only index a dt with an id attribute and an accompanying dd
+                    if not dd or not _id:
+                        continue
+
+                    # Create a copy of the node to avoid manipulating the
+                    # data structure that we're iterating over
+                    dd_copy = HTMLParser(dd.html).body.child
+
+                    # Remove all nested domains from dd_copy.
+                    # They are already parsed separately.
+                    nested_domains = dd_copy.css("dl")
```

Comment:
We need to remove only nested elements that will be indexed (items from description lists that don't have an id won't be indexed).

Reply:
This turned out to be where all the inaccuracy came from :100: Accurately removing everything that will be indexed later should obviously be done in a correct way.

I've gone over a bunch of the test data now and former duplicates, missing texts etc. now seemed to be fixed :+1: 

Thanks for the patience!!